app/src/visionglass/java/com/igalia/wolvic/PlatformActivity.java

```diff
@@ -565,6 +546,28 @@ public void onStopTrackingTouch(SeekBar seekBar) {
                     media.seek((seekBar.getProgress() / 100.0) * media.getDuration());
                 }
             });
+
+            mGestureDetector = new GestureDetector(getApplicationContext(), new GestureDetector.SimpleOnGestureListener() {
+                @Override
+                public boolean onScroll(@Nullable MotionEvent e1, @NonNull MotionEvent e2, float distanceX, float distanceY) {
+                    notifyOnScrollEvent(distanceX, distanceY);
+                    return true;
+                }
+
+                @Override
+                public boolean onSingleTapUp(MotionEvent e) {
```

Comment:
There is a problem when the user does a double tap, which causes the cursor to become stuck in the "pressed" state. This is caused by `onSingleTapUp` calling `touchEvent` with true in the "down" parameter.

Actually, `onSingleTapUp` is probably not needed, since we already use `onSingleTapConfirmed`.

While we are at it, maybe we could also implement `onLongPress` and show the context menu?

Reply:
OK I'll add a handler for double tap to emit the up event too.

We cannot use onDown because all the gestures start with an onDown. So an scroll event would be onDown+onScroll. If you always emit the touchEvent(down) in onDown, every scroll gesture will leave the status inconsistent. And the problem is that you cannot "revert" the previously emitted event in onDown when you receive the onScroll

Regarding the context menu yeah we could do it as well I guess, not sure which events are the engines expecting. Maybe for a folloup patch?